File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        freq = [0] * 26
        for c in s:
            freq[ord(c) - ord('a')] += 1
        for c in t:
            index = ord(c)-ord('a') 
            freq[index] -=1
            if freq[index] < 0:
                return False
        return True
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int[] result = new int[k];
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i:nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        List<Integer>[] bucket = new ArrayList[nums.length+1];
        for(int key:map.keySet()){
            int f = map.get(key);
            if(bucket[f] == null) bucket[f] = new ArrayList<>();
            bucket[f].add(key);
        }
        int index = 0;
        for(int i = bucket.length - 1;i>=0 && index<k;i--){
            if(bucket[i] != null){
                for(int j:bucket[i]){
                    result[index++] = j;
                    if(index == k) break;
                }
            }
        }
        return result;
    }
}
